Paper: aPhyloGeo-Covid: A web interface for phylogeographic analysis of SARS-CoV-2 variation using Neo4j and Snakemake

TahiriNadia opened the pull request:
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.


*Editor*: <!--editor-->@cbcunc<!--end-editor-->

*Reviewers*: <!--reviewers-list-->@cmahapatra97, @EngineerKhan, @ferchaure<!--end-reviewers-list-->

TahiriNadia committed:
abstract

TahiriNadia committed:
intro

TahiriNadia committed:
Acknowledgements

TahiriNadia committed:
References (bib)

TahiriNadia committed:
Methodology

TahiriNadia committed:
Future directions

TahiriNadia committed:
First code

TahiriNadia committed:
Python codes

TahiriNadia committed:
Typo Figure

deniederhut commented:
@scoobies mark paper

deniederhut commented:
@scoobies mark needs review

TahiriNadia committed:
Add Figures 2 and 3

TahiriNadia committed:
Add 3 ref.

TahiriNadia committed:
Add Methodology text

TahiriNadia committed:
Results section

TahiriNadia committed:
Abstract organisation

TahiriNadia committed:
Remove Results Section

TahiriNadia committed:
Results Section

TahiriNadia committed:
Remove Results Section

TahiriNadia committed:
Results Section part 1

TahiriNadia committed:
Update Results Section part 1 (v beta)

TahiriNadia committed:
Update Results Section part 1 (v beta)

TahiriNadia committed:
Remove Results Section

TahiriNadia committed:
Add Figures 2 and 3

TahiriNadia committed:
Update caption Figures

TahiriNadia committed:
Start Results Section

TahiriNadia committed:
Update part 1 Results Section

TahiriNadia committed:
First paragraph Results Section

TahiriNadia committed:
Second Paragraph Results Section

TahiriNadia committed:
Full Results Section

TahiriNadia committed:
Delete figure3.png

TahiriNadia committed:
Add Figure 3

TahiriNadia committed:
Results Section Section v beta

TahiriNadia committed:
Results Section (last paragraph)

TahiriNadia committed:
Remove Å character

TahiriNadia committed:
Work in last paragraph Results Section

TahiriNadia committed:
Results Section last paragraph

TahiriNadia committed:
Remove Å

TahiriNadia committed:
Results Section (in progress)

TahiriNadia committed:
Results Section (in progress)

TahiriNadia committed:
Results Section Final?

TahiriNadia committed:
Results Section (in progress)

TahiriNadia committed:
Results Section (in progress)

TahiriNadia committed:
Results Section (in progress)

TahiriNadia committed:
Results Section (in progress)

TahiriNadia committed:
Two col for Fig 3

TahiriNadia committed:
Scale Figure 2

TahiriNadia committed:
Scale Figure 2

TahiriNadia committed:
Scale Figure 2

TahiriNadia committed:
Scale Figure 2

TahiriNadia committed:
Scale Figure 3

TahiriNadia committed:
Scale Figures 2 and 3

TahiriNadia committed:
Scale Figures 2 and 3

TahiriNadia committed:
Scale Figures 2 and 3

TahiriNadia committed:
Conclusion Section

TahiriNadia committed:
Typo code

WanlinLi2021 committed:
remove two functions

WanlinLi2021 committed:
Update function for lineage table

WanlinLi2021 committed:
adjust code alignment

WanlinLi2021 committed:
adjust code alignment

WanlinLi2021 committed:
adjust code alignment

WanlinLi2021 committed:
adjust code alignment

WanlinLi2021 committed:
adjust code alignment

WanlinLi2021 committed:
Update conclusions

WanlinLi2021 committed:
Update abstract

WanlinLi2021 committed:
Update introduction

TahiriNadia committed:
Corresponding author update

WanlinLi2021 committed:
Update introduction

TahiriNadia committed:
Split the last python code by functions

WanlinLi2021 committed:
Update introduction

WanlinLi2021 committed:
Update introduction

WanlinLi2021 committed:
Update introduction

WanlinLi2021 committed:
Update Introduction

WanlinLi2021 committed:
Update Methodology

TahiriNadia committed:
Reference Figure incomplete

WanlinLi2021 committed:
Update Data Integration

WanlinLi2021 committed:
Update title

WanlinLi2021 committed:
Update title

WanlinLi2021 committed:
Update Input exploration

WanlinLi2021 committed:
Update input explore

WanlinLi2021 committed:
Update subtitles

WanlinLi2021 committed:
Update Lineage-based approach

WanlinLi2021 committed:
Update Lineage-based approach

WanlinLi2021 committed:
Update Location-based approach

WanlinLi2021 committed:
Update code explaination

WanlinLi2021 committed:
Update function explaination

WanlinLi2021 committed:
Update Parameters setting section

WanlinLi2021 committed:
Update  Output exploration section

WanlinLi2021 committed:
Update snakemake section

WanlinLi2021 committed:
Update results section

WanlinLi2021 committed:
Update results

WanlinLi2021 committed:
Update results section

WanlinLi2021 committed:
Update results section

WanlinLi2021 committed:
Update future work section

WanlinLi2021 committed:
Update conclusion section

WanlinLi2021 committed:
Update caption of figure 3

WanlinLi2021 committed:
Update Caption of figure 2

WanlinLi2021 committed:
Update Introduction

WanlinLi2021 committed:
Update conclusions

WanlinLi2021 committed:
Update Introduction

WanlinLi2021 committed:
Delete figure3.png

WanlinLi2021 committed:
add figure 4

WanlinLi2021 committed:
Update figure4

WanlinLi2021 committed:
add figure3

WanlinLi2021 committed:
Update figure 3 caption

WanlinLi2021 committed:
Update figure 3

WanlinLi2021 committed:
Update figure3

WanlinLi2021 committed:
Update figure3

WanlinLi2021 committed:
Delete figure3.png

WanlinLi2021 committed:
add figure3

WanlinLi2021 committed:
Update results

WanlinLi2021 committed:
Update climate data discussion

WanlinLi2021 committed:
Update climate data description

WanlinLi2021 committed:
Update climate description

WanlinLi2021 committed:
Update fig 4 caption

deniederhut commented:
@scoobies assign @cbcunc as editor

scoobies commented:
Assigned! @cbcunc is now the editor

WanlinLi2021 committed:
Delete figure2.png

WanlinLi2021 committed:
Delete figure3.png

WanlinLi2021 committed:
Delete figure4.png

WanlinLi2021 committed:
Add files via upload

WanlinLi2021 committed:
Update figures

WanlinLi2021 committed:
Update figure 2

WanlinLi2021 committed:
Update figure 2

WanlinLi2021 committed:
add figure6

WanlinLi2021 committed:
Delete figure6.png

WanlinLi2021 committed:
add figure6

WanlinLi2021 committed:
Delete figure6.png

WanlinLi2021 committed:
add fig

WanlinLi2021 committed:
Delete figure6.png

WanlinLi2021 committed:
Add files via upload

WanlinLi2021 committed:
Updatefig6 caption

WanlinLi2021 committed:
Update HGT

WanlinLi2021 committed:
Update mybib.bib

WanlinLi2021 committed:
Update HGT

WanlinLi2021 committed:
Update mybib.bib

WanlinLi2021 committed:
Update HGT

WanlinLi2021 committed:
Update HGT results

WanlinLi2021 committed:
Update snakemake discuss

WanlinLi2021 committed:
Update mybib.bib

WanlinLi2021 committed:
Update code

WanlinLi2021 committed:
Update code

WanlinLi2021 committed:
Update function

WanlinLi2021 committed:
Delete figure6.png

WanlinLi2021 committed:
add figures

WanlinLi2021 committed:
Update figure 7

WanlinLi2021 committed:
Update fig6

WanlinLi2021 committed:
Update figure 6

WanlinLi2021 committed:
Update try fig6

WanlinLi2021 committed:
Delete figure3.png

WanlinLi2021 committed:
Delete figure4.png

WanlinLi2021 committed:
Delete figure5.png

WanlinLi2021 committed:
Delete figure6.png

WanlinLi2021 committed:
Delete figure7.png

WanlinLi2021 committed:
add files

WanlinLi2021 committed:
Update figure number

WanlinLi2021 committed:
Update figure caption

WanlinLi2021 committed:
Update figure caption

WanlinLi2021 committed:
Update figure 3 caption

WanlinLi2021 committed:
Update fig 7 ref

WanlinLi2021 committed:
Update rusults

deniederhut commented:
@scoobies add @cmahapatra97 to reviewers

scoobies commented:
@cmahapatra97 added to the reviewers list!

deniederhut commented:
@scoobies code of conduct

scoobies commented:
Our CoC: https://www.scipy2023.scipy.org/code-of-conduct

scoobies commented:
Reports of abusive or harassing behavior may be reported to scipy@enthought.com

cbcunc commented:
Editor says paper is complete and builds. Awaiting review.

deniederhut commented:
@scoobies add @EngineerKhan to reviewers

scoobies commented:
@EngineerKhan added to the reviewers list!

deniederhut commented:
@scoobies add @ferchaure to reviewers

scoobies commented:
@ferchaure added to the reviewers list!

ferchaure commented on papers/nadia_tahiri/00_bibderwalt.rst:
```diff
@@ -0,0 +1,449 @@
+   to explore and visualize a wide range of diverse data sources specifically relevant to  SARS-CoV-2 for phylogeographic analysis. 
+   The integrated Neo4j database acts as a comprehensive repository, consolidating COVID-19 pandemic-related sequences information, 
+   climate data, and demographic data obtained from public databases, facilitating efficient filtering and organization of input data for 
+   phylogeographical studies. Presently, the database encompasses over 113,774 nodes and 194,381 relationships. Once the input dataset is determined. Additionally, aPhyloGeo-Covid provides a scalable and reproducible phylogeographic workflow for investigating the intricate relationship between geographic features and the patterns of variation in diverse SARS-CoV-2 variants. The code repository of ...
```
The sentence "Once the input dataset is determined." like that is unrelated to the rest of the paragraph

TahiriNadia replied:
Thank you, @ferchaure , for your comment and suggestions. We agree, and the appropriate modifications have been incorporated by deleting the sentence.

ferchaure commented on papers/nadia_tahiri/00_bibderwalt.rst:
```diff
@@ -0,0 +1,449 @@
+The challenges associated with data collection, extraction, and integration have hindered the advancement of phylogeographic research within the field. To tackle these challenges, we employed the highly scalable and adaptable Neo4j graph database management system :cite:`guia2017graph` for the storage, management, and querying of extensive SARS-CoV-2 variants-related data. Differing from traditio ...
+
+On the other hand, while recent phylogeographic studies have extensively analyzed the genetic data of species across different geographic regions, many have primarily focused on species distribution or provided visual representations, without investigating the correlation between specific genes (or gene segments) and environmental factors :cite:`uphyrkina2001phylogenetics` :cite:`luo2004phylogeog ...
+:cite:`taylor2020intercontinental` :cite:`aziz2022phylogeography`. To bridge this gap, a novel algorithm applying sliding windows to scan the genetic sequence information related to their climatic conditions was developed by our team :cite:`koshkarov2022phylogeography`. This algorithm utilizes sliding windows to scan genetic sequence information in relation to climatic conditions. Multiple sequen ...
```
maybe the editors know a better way to cite multiple works to get something like: [15-18] that looks better that the current way

TahiriNadia replied:
Once again, thank you for raising this point, with which we agree. While we may not know how to accomplish it ourselves, we believe that @deniederhut t or @cbcunc might possess the knowledge and expertise to do so.

ferchaure commented on papers/nadia_tahiri/00_bibderwalt.rst:
```diff
@@ -0,0 +1,449 @@
+
+A diverse range of data sources pertaining to SARS-CoV-2, covering the period from January 1, 2020, to December 31, 2022, were meticulously extracted, transformed, and loaded into a Neo4j graph database. These sources encompassed SARS-CoV-2 sequences from the SARS-CoV-2 Data Hub :cite:`brister2015ncbi`, lineage development information from Cov-Lineages :cite:`o2021tracking`, population density by ...
+
+To enable efficient querying, configuration of analysis parameters, and output generation within the database, a driver object was established using the Neo4j Python driver to establish seamless connections with the Neo4j database. For phylogeographic analysis, a streamlined workflow was implemented using the Snakemake workflow management system, ensuring an efficient and structured analysis proc ...
```
Here you call it "Plotly-dash" but in other places Dash-Plotly, it's a bit confusing.

WanlinLi2021 replied:
Thank you for bringing this issue to my attention. We replaced "Plotly-dash" with "Dash-Plotly" to maintain consistency throughout the article.

ferchaure commented on papers/nadia_tahiri/00_bibderwalt.rst:
```diff
@@ -0,0 +1,449 @@
+
+Lineage nodes establish connections with Nucleotide and Protein nodes, representing the relationships between lineages and their corresponding genetic sequence data. Moreover, Lineage nodes establish relationships with Location nodes, utilizing the most common occurrence rate as a property. This design empowers researchers to determine the most common countries based on lineage names or search fo ...
+
+.. figure:: figure1.png
```
I know that this is a screenshot... but maybe the font could be increased in a easy way?

WanlinLi2021 replied:
Thank you. We acknowledge that utilizing screenshots does not meet the desired image quality standards, lacks reader-friendliness, and fails to highlight key information effectively. Consequently, we have considered your suggestion and have re-created the images using Drawio to enhance the overall quality.

ferchaure commented on papers/nadia_tahiri/00_bibderwalt.rst:
```diff
@@ -0,0 +1,449 @@
+
+By scanning the complete Multiple Sequence Alignment sequences with a sliding window strategy, the phylogeographic research can effectively focus on sequence information for specific window lengths. To address the integration of genetic and environmental data, complex computational workflows are required, consisting of multiple interdependent processing steps. The aPhyloGeo snakemake workflow add ...
+
+Various tools and software were utilized to accomplish these analysis tasks, including Biopython :cite:`cock2009biopython`, raxml-ng :cite:`kozlov2019raxml`, fasttree :cite:`price2009fasttree`, and Python libraries such as robinson-foulds, NumPy, and Pandas. A manuscript for aPhyloGeo-pipeline is available on Github Wiki (https://github.com/tahiri-lab/aPhyloGeo-pipeline/wiki).
```
Important: add references for Numpy and Pandas

WanlinLi2021 replied:
Thank you for bringing this issue to my attention. I appreciate your feedback. In response to your suggestion, we have now included references for Numpy and Pandas in the article.

ferchaure commented on papers/nadia_tahiri/00_bibderwalt.rst:
```diff
@@ -0,0 +1,449 @@
+
+
+
+.. figure:: figure5.png
```
Did you try with bigger dots or bars? it's tricky see the patters or differences just with little dots far way

WanlinLi2021 replied:
Thank you for your suggestion. We have improved the quality of this figure by increasing the size of the dots.

ferchaure commented on papers/nadia_tahiri/00_bibderwalt.rst:
```diff
@@ -0,0 +1,449 @@
+   
+Our phylogeography-based exploration revealed a notable correlation between mutations in the region [792-940] and precipitation. As a reproducible phylogeographic platform, aPhyloGeo-Covid offers the potential to expand the sample size for further investigation and facilitates the comparability of analysis results.
+
+In addition of correlation analysis between correlation diversity of subfregment of gene and climate condition, we also inferred the ORF1a phylogeny and window regions 792-940 of ORF1a using the RAxML-NG method :cite:`kozlov2019raxml`, and then conducted a detailed horizontal gene transfer-recombination analysis (See Figure :ref:`fig7`) using the HGT-Detection program available on the T-Rex web s ...
```
add somewhere: horizontal gene transfer (HGT)

WanlinLi2021 replied:
Thank you for your suggestion. Yes, it is essential to clarify abbreviations before using them consistently throughout the text. In response to your feedback, we have added "horizontal gene transfer (HGT)" in the sentence where we initially introduced the concept of horizontal gene transfer.

ferchaure commented on papers/nadia_tahiri/00_bibderwalt.rst:
```diff
@@ -0,0 +1,449 @@
+
+The minimum-cost transfer scenario with five HGTs necessary to reconcile the variants and gene phylogenies is shown in Figure :ref:`fig7` (HGTs are depicted by numbered arrows). Prior to the analysis, the Robinson and Foulds distance (RF) was measured at 16, indicating the dissimilarity between the phylogenetic tree of ORF1a and inferred phylogenetic trees (phylogenetic tree of the window regions ...
+In the second iteration, another HGT event was found from subtree QLL26045 to subtrees QPJ77309, QWP89176, and WAV10885, with an RF of 6 and a bipartition dissimilarity of 3.5. After this iteration, the criteria values were RF = 6 and BD = 3.5.
+In the third iteration, an HGT was identified from subtree UJS65740 to subtrees QLL26045, QPJ77309, QPN02498, QWP89176, UJW68561, and WAV10885, with an RF of 4 and a bipartition dissimilarity of 2.5. The criteria values after this iteration remained RF = 4 and BD = 2.5.
```
I personally get confuse with this type of paragraph, maybe a table or something with a structure format could be more clear

WanlinLi2021 replied:
Thank you for the observation. We greatly appreciate it. To ensure conciseness and clarity, we have revised the paragraph concerning transfer-recombination analysis (HGT). Additionally, we have included a table to enhance the clarity of the information presented.

ferchaure reviewed (changes requested):
Hi Nadia, Amazing and really interesting work.

The only required change is adding the reference for Numpy and Pandas.
I added a few additional comments 


Cheers,
Fer

EngineerKhan commented:
Same thoughts as @ferchaure. I am not an expert on the topic, so my opinion should be taken with a grain of salt. Here are some minor suggested changes, though:

- In the abstract, "a wide range of" is unnecessary.
- "Throughout the COVID-19 pandemic, researchers......" – this is a new section and can be in a new paragraph.
- "Climate data can be obtained from reputable sources such as NASA/POWER DailyGridded weather" – climate data of? Can add more clarity.
- "The challenges associated with data collection, extraction, and integration have hindered the advancement of phylogeographic research within the field. " – I think a reference can be added to this statement.
- "To tackle these challenges, we employed....." – it can also be taken in a new paragraph.
- "To address these challenges, we devised a phylogeographic...." should also be a new paragraph.
- Reference 26 needs to be improved to reflect the better citation of Pandas, like [this](https://conference.scipy.org/proceedings/scipy2010/pdfs/mckinney.pdf).
- "In the subsequent phylogeographical analysis phase, our modularized Snakemake workflow is utilized to
examine how genetic variation patterns within different SARS-CoV-2 variants align with geographic features." – Comment: This line sounds unnecessary for the introduction. Feel free to disagree.
- "from the SARS-CoV-2 Data Hub [7]" – is it the correct reference?
- "population density by country, positivity rates, vaccination rates, diabetes rates, aging data" – sounds like too much info for a general reader.
- " Moreover, the interactive visualization capabilities offered by the Plotly-dash library" – Suggestion: we can add the Plotly-dash library's reference as well.
- The first paragraph of "Data Integration" section contains a lot of attributes and can have a table to complement it.
- "An interactive platform using Dash-Plotly [27] was developped" – please mind the typo.
- "The integration of the Dash platform with the Neo4j graph database allows for seamless retrieval of pertinent data from interconnected nodes based on user-provided keywords related to lineages or locations." – minor: "the" before "seamless retrieval."
- "The aPhyloGeo-Covid offers two distinct approaches for selecting input datasets" – Suggestion: I think these two options would be better off in 2 subheadings rather than 2 numbered sentences.
- "Subsequently, the Neo4j database was queried to identify lineages" – "was" or "is"?
- QUERY: Can we caption the code snippets? @scoobies 
- Suggestion: Fig 2 and 3 can be separated.
- "and the Neighbor-Joining clustering method [31]. " – don't think [31] is the right reference for that method.


Some general observations:
- Majority of references are review/comparative studies.
- At times, it gets verbose.
- Too many figures (numbers) within the text. Some of them can be better represented in tables.
- Is page limit (ref included) 8 or 9? @scoobies

cmahapatra97 commented:
Hi Nadia, 

Very interesting tool development to study SARS-CoV-2 variation. 
I suggest adding some lines on the pros of this model regarding other developed tools based on Neo4J. For example see Laddada W, Zanni-Merk C, Soualmia LF. Analyzing SARS-CoV-2 Sequence Patterns by Semantic Trajectories. Stud Health Technol Inform. 2022 Jun 29;295:197-200. doi: 10.3233/SHTI220696. PMID: 35773842.

Cheers,
Chit

TahiriNadia committed:
Update 00_bibderwalt.rst

Typo text removed.

WanlinLi2021 committed:
Delete figure1.png

WanlinLi2021 committed:
update figure1

WanlinLi2021 committed:
Delete figure1.png

WanlinLi2021 committed:
update figure1

WanlinLi2021 committed:
Replace "Plotly-dash" with "Dash-Plotly" to make the article consistent.

WanlinLi2021 committed:
Update mybib.bib add references for Numpy and Pandas

WanlinLi2021 committed:
add references for Numpy and Pandas

WanlinLi2021 committed:
add horizontal gene transfer (HGT)

WanlinLi2021 committed:
In the abstract, remove "a wide range of" which is unnecessary.

WanlinLi2021 committed:
separate "Throughout the COVID-19 pandemic, researchers......"  in a new paragraph.

WanlinLi2021 committed:
add more clarity for  "Climate data....NASA/POWER DailyGridded

WanlinLi2021 committed:
Update mybib.bib for the topic of challenges associated with data collection

WanlinLi2021 committed:
Add references for topic of data collection challenges

WanlinLi2021 committed:
separate contents in new paragraphs to enhance clarity

WanlinLi2021 committed:
Update mybib.bib for neighbor-joining

WanlinLi2021 committed:
update reference for Neighnor-joining

WanlinLi2021 committed:
Fix grammatical errors

"Subsequently, the Neo4j database was queried to identify lineages" 

“was" ---> "is"

WanlinLi2021 committed:
fix the the typo 

"An interactive platform using Dash-Plotly [27] was developped" – please mind the typo.

"developped" ----> "developed"

WanlinLi2021 committed:
add reference for SARS_CoV-2 Data Hub

WanlinLi2021 committed:
Update reference for SARS-CoV-2 Data Hub

WanlinLi2021 committed:
add reference for Dash-Plotly

WanlinLi2021 committed:
Update mybib.bib for Dash-Plotly

WanlinLi2021 committed:
add references for Dash-Plotly

WanlinLi2021 committed:
Revise verbose paragraph to make them clearer

WanlinLi2021 committed:
add Pandas ref

WanlinLi2021 committed:
add Pandas ref

WanlinLi2021 committed:
Update horizontal gene transfer (HGT) acronym

WanlinLi2021 committed:
add detail for climate data (Introduction)

TahiriNadia committed:
update key of Pandas paper

TahiriNadia committed:
update key ref

WanlinLi2021 committed:
add "the" before "seamless retrieval

TahiriNadia committed:
convert list of approaches in subheadings

WanlinLi2021 committed:
Update ref for Neighbor-Joining

WanlinLi2021 committed:
Update Data Integration

WanlinLi2021 committed:
Add table 1 for data integration section

WanlinLi2021 committed:
Update table 1

WanlinLi2021 committed:
test multiple row in table

WanlinLi2021 committed:
test table1 line break

WanlinLi2021 committed:
test table 1 line break

WanlinLi2021 committed:
remove table

WanlinLi2021 committed:
add table 1

WanlinLi2021 committed:
add table 2

WanlinLi2021 committed:
Update table ref

TahiriNadia committed:
update figure

WanlinLi2021 committed:
Update data sources description

TahiriNadia committed:
option table 1

WanlinLi2021 committed:
Update table 2

TahiriNadia commented:
@scoobies build paper

scoobies commented:
building paper...

scoobies commented:
{"data"=>{"info"=>"Build for paper tahiri-lab-2023 scheduled.  Note that builds are only executed if the current build attempt is more than 5 minutes old."}, "status"=>"success"}

WanlinLi2021 committed:
test

WanlinLi2021 committed:
test

WanlinLi2021 committed:
test

WanlinLi2021 committed:
test

WanlinLi2021 committed:
remove test table

WanlinLi2021 committed:
Delete figure5.png

WanlinLi2021 committed:
update figure 5

TahiriNadia committed:
Update table 1

TahiriNadia committed:
update table 1

TahiriNadia committed:
update table 1

TahiriNadia committed:
update option table 1

TahiriNadia committed:
add table test

TahiriNadia committed:
update table 1

TahiriNadia committed:
+ large table 1

TahiriNadia committed:
Update table 1

TahiriNadia committed:
table 1 options

TahiriNadia committed:
table 2

TahiriNadia committed:
table 1 version works

TahiriNadia committed:
table 1

TahiriNadia committed:
table 1

TahiriNadia committed:
table 1

TahiriNadia committed:
table 1

TahiriNadia committed:
figure 5 size

TahiriNadia committed:
figure 5

WanlinLi2021 committed:
reply to reviewer 3

WanlinLi2021 committed:
add ref Laddada 2022

TahiriNadia committed:
table 2

TahiriNadia committed:
table 2

TahiriNadia committed:
table 2

TahiriNadia committed:
table 2

TahiriNadia committed:
table 2

WanlinLi2021 committed:
add note to the table 2

TahiriNadia committed:
table 2

TahiriNadia committed:
table 2

TahiriNadia committed:
typo

TahiriNadia committed:
typo

TahiriNadia commented:
> Same thoughts as @ferchaure. I am not an expert on the topic, so my opinion should be taken with a grain of salt. Here are some minor suggested changes, though:
> 
> * In the abstract, "a wide range of" is unnecessary.
> * "Throughout the COVID-19 pandemic, researchers......" – this is a new section and can be in a new paragraph.
> * "Climate data can be obtained from reputable sources such as NASA/POWER DailyGridded weather" – climate data of? Can add more clarity.
> * "The challenges associated with data collection, extraction, and integration have hindered the advancement of phylogeographic research within the field. " – I think a reference can be added to this statement.
> * "To tackle these challenges, we employed....." – it can also be taken in a new paragraph.
> * "To address these challenges, we devised a phylogeographic...." should also be a new paragraph.
> * Reference 26 needs to be improved to reflect the better citation of Pandas, like [this](https://conference.scipy.org/proceedings/scipy2010/pdfs/mckinney.pdf).
> * "In the subsequent phylogeographical analysis phase, our modularized Snakemake workflow is utilized to
>   examine how genetic variation patterns within different SARS-CoV-2 variants align with geographic features." – Comment: This line sounds unnecessary for the introduction. Feel free to disagree.
> * "from the SARS-CoV-2 Data Hub [7]" – is it the correct reference?
> * "population density by country, positivity rates, vaccination rates, diabetes rates, aging data" – sounds like too much info for a general reader.
> * " Moreover, the interactive visualization capabilities offered by the Plotly-dash library" – Suggestion: we can add the Plotly-dash library's reference as well.
> * The first paragraph of "Data Integration" section contains a lot of attributes and can have a table to complement it.
> * "An interactive platform using Dash-Plotly [27] was developped" – please mind the typo.
> * "The integration of the Dash platform with the Neo4j graph database allows for seamless retrieval of pertinent data from interconnected nodes based on user-provided keywords related to lineages or locations." – minor: "the" before "seamless retrieval."
> * "The aPhyloGeo-Covid offers two distinct approaches for selecting input datasets" – Suggestion: I think these two options would be better off in 2 subheadings rather than 2 numbered sentences.
> * "Subsequently, the Neo4j database was queried to identify lineages" – "was" or "is"?
> * QUERY: Can we caption the code snippets? @scoobies
> * Suggestion: Fig 2 and 3 can be separated.
> * "and the Neighbor-Joining clustering method [31]. " – don't think [31] is the right reference for that method.
> 
> Some general observations:
> 
> * Majority of references are review/comparative studies.
> * At times, it gets verbose.
> * Too many figures (numbers) within the text. Some of them can be better represented in tables.
> * Is page limit (ref included) 8 or 9? @scoobies


Thank you, @EngineerKhan, for your comment and suggestions. Below are the answers to your questions.


✅ In the abstract, "a wide range of" is unnecessary. 

Thank you for your suggestion. We agree that the phrase "a wide range of" is redundant. Therefore, we have removed it from the text. 

✅ "Throughout the COVID-19 pandemic, researchers......" – this is a new section and can be in a new paragraph. 

We appreciate the suggestion. We also believe that separating it into a new section will enhance clarity. We have taken your advice and incorporated the content into a new section. 

✅ "Climate data can be obtained from reputable sources such as NASA/POWER DailyGridded weather" – climate data of? Can add more clarity. 

Thank you for the suggestion. In order to enhance clarity when referring to the climate data, we have optimized the sentence as follows: "Daily climate data (e.g., humidity, wind speed, and temperature) for each location involved during the pandemic can be obtained from reputable sources such as NASA/POWER DailyGridded Weather." This revision aims to provide more clarity and guidance to readers. 

✅ "The challenges associated with data collection, extraction, and integration have hindered the advancement of phylogeographic research within the field. " – I think a reference can be added to this statement. 

Thank you for your suggestion. We greatly appreciate it. We have taken action by including two additional references to substantiate the information presented. 

✅ "To tackle these challenges, we employed....." – it can also be taken in a new paragraph. 

We appreciate the suggestion. We also believe that separating it into a new section will enhance clarity. We have taken your advice and incorporated the content into a new section, making it easier for readers to locate and understand. 

✅ "To address these challenges, we devised a phylogeographic...." should also be a new paragraph. 

We appreciate the suggestion. We have taken your advice and incorporated the content into a new section. 

✅ Reference 26 needs to be improved to reflect the better citation of Pandas, like [this](https://conference.scipy.org/proceedings/scipy2010/pdfs/mckinney.pdf). 

Thank you for bringing this issue to our attention. We appreciate your feedback. In response to your suggestion, we have now included this reference for Pandas in the article. 

✅ "In the subsequent phylogeographical analysis phase, our modularized Snakemake workflow is utilized to examine how genetic variation patterns within different SARS-CoV-2 variants align with geographic features." – Comment: This line sounds unnecessary for the introduction. Feel free to disagree. 

Thank you for your suggestion. Upon careful consideration, we have decided to retain this line, as it offers a comprehensive overview of the Snakemake section. We believe that including this line will assist readers in understanding the Snakemake portion before delving into the Methodology section. 

✅ "from the SARS-CoV-2 Data Hub [7]" – is it the correct reference? 

Thanks for the reminder. We agree and have added the reference given directly to the SARS-CoV-2 Data Hub website (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/docs/participate/#how-to-cite-us). 

✅ "population density by country, positivity rates, vaccination rates, diabetes rates, aging data" – sounds like too much info for a general reader. 

Thank you for bringing this issue to our attention. In order to facilitate comprehension for general readers, we have divided the long sentences into concise lists. Additionally, we have included Table 1 to provide a clear and accessible representation of the information for easy understanding. 

✅ " Moreover, the interactive visualization capabilities offered by the Plotly-dash library" – Suggestion: we can add the Plotly-dash library's reference as well. 

I sincerely appreciate your suggestion. In response, we have added two references to support further the information provided. One of the references aligns with the suggestion provided by the Plotly Community (source: https://community.plotly.com/t/citing-plotly-and-dash-in-academic-papers/33045). Additionally, we have included a review of Dynamic Dashboards, which include the Dash-Plotly component. 

✅ The first paragraph of "Data Integration" section contains a lot of attributes and can have a table to complement it. 

Thank you for your suggestion. In order to enhance clarity and facilitate understanding, we have incorporated a table into this section. 

✅ "An interactive platform using Dash-Plotly [27] was developped" – please mind the typo. 

Thank you very much for the reminder. We apologize for this mistake. We should have used "developed". We have changed it accordingly. Thanks again. 

✅ "The integration of the Dash platform with the Neo4j graph database allows for seamless retrieval of pertinent data from interconnected nodes based on user-provided keywords related to lineages or locations." – minor: "the" before "seamless retrieval." 

Thank you for this note. We added "the" before "seamless retrieval." 

✅ "The aPhyloGeo-Covid offers two distinct approaches for selecting input datasets" – Suggestion: I think these two options would be better off in 2 subheadings rather than 2 numbered sentences. 

Thank you for this observation. We updated the numbered sentences to subheadings. 

✅ "Subsequently, the Neo4j database was queried to identify lineages" – "was" or "is"? 

Thank you very much for the correction. In the corresponding context in the text, we should have used "is". We have changed it accordingly. Thanks again. 

✅ QUERY: Can we caption the code snippets? [@scoobies](https://github.com/scoobies) 

Thanks for this note and observation. We use exactly the same template given by the SciPy conference (source: https://github.com/scipy-conference/scipy_proceedings/blob/2023/papers/00_bibderwalt/00_bibderwalt.rst) and the template has no caption for code snippets. Thanks. 

 

✅ Suggestion: Fig 2 and 3 can be separated. 

Thank you for your feedback. During the PDF generation process, the rst compiler can automatically reposition figures and tables to optimize spacing. Originally, figure 2 and figure 3 are separated in our Github text. 

✅ "and the Neighbor-Joining clustering method [31]. " – don't think [31] is the right reference for that method. 

Thank you for the note. We understand and agree that the original authors of Neighbor-Joining are Saitou and Nei in 1987(Saitou, Naruya, and Masatoshi Nei, "The neighbor-joining method: a new method for reconstructing phylogenetic trees," Molecular Biology and Evolution,1987). This revised reference will provide a more authoritative and reliable source of information for the topic at hand. 

 

Some general observations: 

✅ Majority of references are review/comparative studies. 

We have added new original references to counter the review/comparative studies. Thanks. 

✅ At times, it gets verbose.  

Thank you for your feedback. We have taken action by rewording certain sentences, splitting others in half, and incorporating additional details and tables. 

✅ Too many figures (numbers) within the text. Some of them can be better represented in tables. 

We have added new tables, and the figures in the text are now easier to understand. Thank you for your attention. 

✅ Is page limit (ref included) 8 or 9? [@scoobies](https://github.com/scoobies) 

Thank you for the reminder. We confirm that the page limit is 8, excluding references. The SciPy conference rule ‘The compiled version of the paper (PDF) should be at most 8 pages, including figures but not including references.’ (Source: https://github.com/scipy-conference/scipy_proceedings/tree/2023).

TahiriNadia commented:
> Hi Nadia,
> 
> Very interesting tool development to study SARS-CoV-2 variation. I suggest adding some lines on the pros of this model regarding other developed tools based on Neo4J. For example see Laddada W, Zanni-Merk C, Soualmia LF. Analyzing SARS-CoV-2 Sequence Patterns by Semantic Trajectories. Stud Health Technol Inform. 2022 Jun 29;295:197-200. doi: 10.3233/SHTI220696. PMID: 35773842.
> 
> Cheers, Chit

Thank you, @cmahapatra97, for recommending this literature. The article is impressive with the application of Graph theory to mutation tracking. It highlights the advantages of Neo4j in relation analysis. Therefore, we have added a related discussion in the Introduction section.

TahiriNadia committed:
typo

cbcunc commented:
@scoobies mark pending comment

EngineerKhan commented:
Thanks a lot, @TahiriNadia for taking my suggestions positively. I was away on vacation, so apologies for the late reply/verification.

cbcunc commented:
@EngineerKhan does this paper meet your final review approval?

cbcunc commented:
@scoobies check references

scoobies commented:
Checking the BibTeX entries failed with the following error: 
```
Lexical or syntactical errors: 

@article{khan2021structurestitle={Structures of SARS-CoV-2 RNA-binding proteins and therapeutic targets},author={Khan, Muhammad Tahir and Irfan, Muhammad and Ahsan, Hina and Ahmed, Abrar and Kaushik, Aman Chandra and Khan, Anwar Sheed and Chinnasamy, Sathishkumar and Ali, Arif and Wei, Dong-Qing},journal={Intervirology},volume={64},number={2},pages={55--68},year={2021},publisher={Karger Publishers},doi=“
@article{ni2023structuraltitle={Structural insights into ribonucleoprotein dissociation by nucleocapsid protein interacting with non-structural protein 3 in SARS-CoV-2},author={Ni, Xincheng and Han, Yinze and Zhou, Renjie and Zhou, Yanmei and Lei, Jian},journal={Communications Biology},volume={6},number={1},pages={193},year={2023},publisher={Nature Publishing Group UK London},doi=“
@article{waterhouse2018swisstitle={SWISS-MODEL: homology modelling of protein structures and complexes},author={Waterhouse, Andrew and Bertoni, Martino and Bienert, Stefan and Studer, Gabriel and Tauriello, Gerardo and Gumienny, Rafal and Heer, Florian T and de Beer, Tjaart A P and Rempfer, Christine and Bordoli, Lorenza and others},journal={Nucleic acids research},volume={46},number={W1},pages={W296--W303},year={2018},publisher={Oxford University Press},doi=“
```

cbcunc commented:
@TahiriNadia Would you please fix the above BibTex DOI errors? I think they need to go inside curly braces rather than quotes?

cbcunc commented:
@scoobies check references

scoobies commented:
Checking the BibTeX entries failed with the following error: 
```
Lexical or syntactical errors: 

@article{khan2021structurestitle={Structures of SARS-CoV-2 RNA-binding proteins and therapeutic targets},author={Khan, Muhammad Tahir and Irfan, Muhammad and Ahsan, Hina and Ahmed, Abrar and Kaushik, Aman Chandra and Khan, Anwar Sheed and Chinnasamy, Sathishkumar and Ali, Arif and Wei, Dong-Qing},journal={Intervirology},volume={64},number={2},pages={55--68},year={2021},publisher={Karger Publishers},doi=“
@article{ni2023structuraltitle={Structural insights into ribonucleoprotein dissociation by nucleocapsid protein interacting with non-structural protein 3 in SARS-CoV-2},author={Ni, Xincheng and Han, Yinze and Zhou, Renjie and Zhou, Yanmei and Lei, Jian},journal={Communications Biology},volume={6},number={1},pages={193},year={2023},publisher={Nature Publishing Group UK London},doi=“
@article{waterhouse2018swisstitle={SWISS-MODEL: homology modelling of protein structures and complexes},author={Waterhouse, Andrew and Bertoni, Martino and Bienert, Stefan and Studer, Gabriel and Tauriello, Gerardo and Gumienny, Rafal and Heer, Florian T and de Beer, Tjaart A P and Rempfer, Christine and Bordoli, Lorenza and others},journal={Nucleic acids research},volume={46},number={W1},pages={W296--W303},year={2018},publisher={Oxford University Press},doi=“
```

cbcunc commented:
@TahiriNadia We figured out what's wrong with the DOIs. They are quoted with smart quotes instead of straight quotes. Would you please this? We are having a final review today and this is one of the few papers which are not ready.

WanlinLi2021 committed:
Fix Lexical or syntactical errors 

They are quoted with smart quotes instead of straight quotes.

WanlinLi2021 commented:
> @TahiriNadia We figured out what's wrong with the DOIs. They are quoted with smart quotes instead of straight quotes. Would you please this? We are having a final review today and this is one of the few papers which are not ready.

Thank you very much for the reminder. Sorry for not responding in time. I have made corrections to the QUOTES for these references.

cbcunc commented:
@scoobies check references

scoobies commented:
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1111/mec.15222 is OK
- 10.1073/pnas.2218012120 is OK
- 10.1111/tbed.14556 is OK
- 10.1111/mec.15991 is OK
- 10.1046/j.1365-294X.2003.02051.x is OK
- 10.12688/wellcomeopenres.16661.2 is OK
- 10.1093/nar/gku1207 is OK
- 10.46234/ccdcw2021.255 is OK
- 10.1016/j.heliyon.2021.e06625 is OK
- 10.1038/s41562-021-01122-8 is OK
- 10.1109/ICDEW.2012.31 is OK
- 10.5220/0006356003510356 is OK
- 10.1046/j.0962-1083.2001.01350.x is OK
- 10.1371/journal.pbio.0020442 is OK
- 10.1038/s41598-020-58743-8 is OK
- 10.3354/esr01188 is OK
- 10.25080/majora-212e5952-018 is OK
- 10.1016/0025-5564(81)90043-2 is OK
- 10.1093/bioinformatics/bty350 is OK
- 10.1093/nar/gkaa434 is OK
- 10.1016/j.future.2023.01.009 is OK
- 10.1038/s41592-021-01254-9 is OK
- 10.1093/bioinformatics/btp163 is OK
- 10.3153/AR19009 is OK
- 10.1145/1900008.1900067 is OK
- 10.1093/database/baab026 is OK
- 10.1007/978-1-4842-0022-3_9 is OK
- 10.1093/molbev/mst010 is OK
- 10.1016/j.sbi.2006.04.004 is OK
- 10.1007/s00453-007-9116-4 is OK
- 10.1093/oxfordjournals.molbev.a040454 is OK
- 10.1093/bioinformatics/btz305 is OK
- 10.1093/molbev/msp077 is OK
- 10.1007/978-3-030-78829-2_9 is OK
- 10.1159/000513686 is OK
- 10.1038/s42003-023-04570-2 is OK
- 10.1093/nar/gky427 is OK
- 10.1093/nar/gks485 is OK
- 10.1016/S0092-8674(00)00175-6 is OK
- 10.1186/s12862-020-01732-2 is OK
- 10.1093/nar/gkr735 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1007/978-1-4842-0241-8_5 is OK
- 10.1111/jbi.14481 is OK
- 10.1016/j.ympev.2011.12.007 is OK
- 10.1093/nar/gkw1065 is OK
- 10.25080/Majora-7ddc1dd1-012 is OK
- 10.25080/Majora-92bf1922-00a is OK
- 10.3233/SHTI220696 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

cbcunc commented:
@scoobies mark ready

TahiriNadia commented:
> @TahiriNadia Would you please fix the above BibTex DOI errors? I think they need to go inside curly braces rather than quotes?

Thank you, @cbcunc.

deniederhut reviewed (approved):
Thanks @cmahapatra97, @EngineerKhan, and @ferchaure for your help getting this paper ready for publication! And thanks to @TahiriNadia for joining us at SciPy this year (and many other years)!!!

![giphy-1806883251](https://github.com/scipy-conference/scipy_proceedings/assets/6868396/cffb5145-9c50-4145-b0b8-8408ff024e67)
